libImaging/ZipDecode.c

```diff
@@ -274,4 +279,20 @@ ImagingZipDecode(Imaging im, ImagingCodecState state, UINT8* buf, int bytes)
 
 }
 
+
+int ImagingZipDecodeCleanup(ImagingCodecState state){
+    /* called to fee the decompression engine when the decode terminates
```

Comment:
Hmm. I went ahead and created #2542 before realising you'd bundled it into this PR. I'll leave the PR open and see what happens, if you don't mind, just because it's an independent change.